File: src/lessons/recursion/DragonCurve1Entity.java
package lessons.recursion;

public class DragonCurve1Entity extends jlm.universe.turtles.Turtle {

	/* BEGIN TEMPLATE */
public void dragon(int ordre, double x, double y, double z, double t) {
		/* BEGIN SOLUTION */
		double u, v;

		if (ordre == 1) {
			setPos(x, y);
			moveTo(z, t);
		} else {
			u = (x + z + t - y) / 2;
			v = (y + t - z + x) / 2;
			dragon(ordre - 1, x, y, u, v);
			dragon(ordre - 1, z, t, u, v);
		}
		/* END SOLUTION */
}
	/* END TEMPLATE */

	public void run() {
		dragon((Integer) getParam(0), (Double) getParam(1), (Double) getParam(2), (Double) getParam(3),
				(Double) getParam(4));
	}

}

File: src/lessons/recursion/dragoncurve1/DragonCurve1Entity.py
# BEGIN TEMPLATE
def dragon(ordre, x, y, z, t):
  # BEGIN SOLUTION 
    if ordre == 1):
      setPos(x, y)
      moveTo(z, t)
    else:
      u = (x + z + t - y) / 2
      v = (y + t - z + x) / 2;
      dragon(ordre - 1, x, y, u, v)
      dragon(ordre - 1, z, t, u, v)
# END SOLUTION
# END TEMPLATE

dragon(getParam(0), getParam(1), getParam(2), getParam(3), getParam(4))
